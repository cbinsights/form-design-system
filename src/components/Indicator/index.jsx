import React from 'react';
import PropTypes from 'prop-types';
import cc from 'classcat';

export const VALID_TYPES = ['update', 'count'];

const Indicator = ({ label, type = 'update' }) => {
  const isUpdate = type === 'update';

  return (
    <div className="fds-indicatorWrapper alignChild--center--center">
      <div
        aria-label={isUpdate ? 'Notification indicator' : 'Count indicator'}
        className={cc([
          'fds-indicator',
          'padding--x--xs fontSize--xs alignChild--center--center',
          {
            'shape--circle fds-indicator--dot color--white': !label,
            'color--white bgColor--red': isUpdate,
            'color--aqua': !isUpdate,
            'bgColor--aqua': !isUpdate && !label,
          },
        ])}
      >
        <span className="fds-indicator-label wrap--singleLine--truncate">{label}</span>
      </div>
    </div>
  );
};

Indicator.propTypes = {
  /**
   * String to render inside the Indicator bubble.
   * The `element` type should be reserved for FDS icons.
   */
  label: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),

  /** Type of Indicator */
  type: PropTypes.oneOf(VALID_TYPES),
};

export default Indicator;
