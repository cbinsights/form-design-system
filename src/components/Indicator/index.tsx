import React from 'react';
import cc from 'classcat';

export interface IndicatorProps {
  /** Text inside the button */
  label: string;
  /** Controls size display */
  size?: 'xs' | 's';
}

const Indicator = ({ label, size = 's', ...rest }: IndicatorProps): JSX.Element => (
  <div
    {...rest}
    className={cc(['fdsIndicator', 'wrap--singleLine', `fdsIndicator--${size}`])}
  >
    {label}
  </div>
);

export default Indicator;
