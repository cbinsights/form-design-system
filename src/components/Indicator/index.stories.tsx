import { StoryObj } from '@storybook/react';
import Indicator, { IndicatorProps } from '.';

export const Primary: StoryObj<IndicatorProps> = {
  args: {
    label: 'Indicator',
  },
};

export default {
  component: Indicator,
  title: 'components/Indicator',
  parameters: {
    docs: {
      fds: {
        attributes: ['rest'],
      },
    },
  },
};
