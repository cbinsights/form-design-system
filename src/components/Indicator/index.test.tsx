import React from 'react';
import { shallow } from 'enzyme';

import Indicator from '.';

describe('Indicator component', () => {
  it('matches snapshot (default props)', () => {
    const component = shallow(<Indicator label="badge" />);
    expect(component).toMatchSnapshot();
  });

  it('matches snapshot (set all props)', () => {
    const component = shallow(<Indicator label="badge" size="xs" data-test="test" />);
    expect(component).toMatchSnapshot();
  });
});
