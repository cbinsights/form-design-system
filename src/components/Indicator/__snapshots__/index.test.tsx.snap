// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indicator component matches snapshot (default props) 1`] = `
<div
  className="fdsIndicator wrap--singleLine fdsIndicator--s"
>
  badge
</div>
`;

exports[`Indicator component matches snapshot (set all props) 1`] = `
<div
  className="fdsIndicator wrap--singleLine fdsIndicator--xs"
  data-test="test"
>
  badge
</div>
`;
